src/topp/OMSSAAdapter.cpp

```diff
@@ -1018,6 +1019,33 @@ class TOPPOMSSAAdapter :
     protein_identification.setSearchEngineVersion(omssa_version);
     protein_identification.setSearchEngine("OMSSA");
 
+    // reannotate file oriin and native ids
+    MSExperiment exp;
+    MzMLFile().load(inputfile_name, exp);
+    protein_identification.setPrimaryMSRunPath({inputfile_name}, exp);
+
+    // add RT and precursor m/z to the peptide IDs (look them up in the spectra):
+    SpectrumLookup lookup;
+    lookup.readSpectra(exp);
+
+    for (auto& p : peptide_ids)
```

Comment:
this for loop can be refactored into a member of SpectrumLookup?!
I think there are at least 2 copies of this code in place, which we could get rid of.
I can do that

Reply:
this would be nice! thanks